autosklearn/ensemble_building/builder.py

```diff
@@ -745,10 +749,10 @@ def fit_ensemble(
         *,
         size: int | None = None,
         task: int | None = None,
-        metric: Scorer | None = None,
+        metrics: Sequence[Scorer] | None = None,
         precision: int | None = None,
         random_state: int | np.random.RandomState | None = None,
-    ) -> EnsembleSelection:
+    ) -> AbstractEnsemble:
```

Comment:
Not important but just a note on typing: Usually it's best to have precise types for what is returned, i.e. `EnsembleSelection` or `MultiObjectiveEnsembleWrapper`.

If the metrics must always be passed in as a parameter, this is possible to do with `@overload`, otherwise when it's optionally None, and we default to what's passed in `__init__`, it's doable but a lot more ugly.

```python
@overload
def fit_ensemble(
    # Other params
    metrics: Scorer
) -> EnsembleSelection:
    ...
    
@overload
def fit_ensemble(
    # Other params
    metrics: Sequence[Scorer]
) -> MultiObjectiveEnsembleWrapper:
    ...
    
def fit_ensemble(
    # other params
    metrics: Scorer | Sequence[Scorer]
) -> EnsembleSelection | MultiObjectiveEnsembleWrapper:
    # actual implementation
    ... 
```

Also note that it's only possible if we have `Scorer | Sequence[Scorer]` instead of always `Sequence[Scorer]`.

Again, not important, just sharing some info

Reply:
Honestly, I don't know why we can pass in the `metrics` here again... 

I'll keep the current annotation of returning `AbstractEnsemble` for forward compatibility with the user being able to pass in any kind of ensemble builder.